Build failure: libretro.parallel-n64

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

use nixos-unstable-small enable global cudaSupport on nixpkgs

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

  hydra search doesn't work for me. 500

### Relevant log output

```console
from mupen64plus-core/src/r4300/cached_interp.c:37:
mupen64plus-core/src/r4300/hacktarux_dynarec/assemble.h:1653:20: note: previous definition of 'fsqrt' with type 'void(void)'
 1653 | static INLINE void fsqrt(void)
      |                    ^~~~~
gcc  -DARCH_MIN_SSE2 -DHAVE_RICE -DHAVE_GLN64 -DHAVE_GLIDE64 -DHAVE_THR_AL  -DHAVE_PARALLEL -DHAVE_PARALLEL_RSP -DPARALLEL_INTEGRATION -I./mupen64plus-rsp-paraLL>
gcc  -DARCH_MIN_SSE2 -DHAVE_RICE -DHAVE_GLN64 -DHAVE_GLIDE64 -DHAVE_THR_AL  -DHAVE_PARALLEL -DHAVE_PARALLEL_RSP -DPARALLEL_INTEGRATION -I./mupen64plus-rsp-paraLL>
gcc  -DARCH_MIN_SSE2 -DHAVE_RICE -DHAVE_GLN64 -DHAVE_GLIDE64 -DHAVE_THR_AL  -DHAVE_PARALLEL -DHAVE_PARALLEL_RSP -DPARALLEL_INTEGRATION -I./mupen64plus-rsp-paraLL>
In file included from mupen64plus-core/src/r4300/fpu.h:25,
                 from mupen64plus-core/src/r4300/mips_instructions.def:50,
                 from mupen64plus-core/src/r4300/pure_interp.c:179:
/nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/bits/mathcalls-narrow.h:36:20: error: conflicting types for 'fsqrt'; have 'float(double)'
   36 | __MATHCALL_NARROW (__MATHCALL_NAME (sqrt), __MATHCALL_REDIR_NAME2 (sqrt), 1);
      |                    ^~~~~~~~~~~~~~~
In file included from mupen64plus-core/src/r4300/recomp.h:49,
                 from mupen64plus-core/src/r4300/cached_interp.h:30,
                 from mupen64plus-core/src/r4300/pure_interp.c:37:
mupen64plus-core/src/r4300/hacktarux_dynarec/assemble.h:1653:20: note: previous definition of 'fsqrt' with type 'void(void)'
 1653 | static INLINE void fsqrt(void)
      |                    ^~~~~
make: *** [Makefile:1069: mupen64plus-core/src/r4300/cached_interp.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [Makefile:1069: mupen64plus-core/src/r4300/pure_interp.o] Error 1
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2-cachyos, NixOS, 26.05 (Yarara), 26.05.20251228.c0b0e0f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - nixpkgs: `/nix/store/xzl5dhddgjxp2p18h5qpkhvlkp8128j1-source`



### Notify maintainers

@aanderse @edwtjo @hrdinka @thiagokokada

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/emulators/libretro/cores/parallel-n64.nix
+++ b/pkgs/applications/emulators/libretro/cores/parallel-n64.nix
@@ -18,6 +18,14 @@ mkLibretroCore {
     hash = "sha256-Th8VqENewfTeRTH+lONN7ZTMLQ0G6901q6ZBNMgepL4=";
   };
 
+  patches = [
+    # Fix build with gcc15
+    # Upstream considers this core legacy (for "potato PC") and won't fix.
+    # See: https://github.com/libretro/parallel-n64/issues/797
+    #   /nix/store/...-glibc-2.40-66-dev/include/bits/mathcalls-narrow.h:36:20: error: conflicting types for 'fsqrt'; have 'float(double)'
+    ./patches/parallel-n64-gcc15.patch
+  ];
+
   extraBuildInputs = [
     libGLU
     libGL

--- a/pkgs/applications/emulators/libretro/cores/patches/parallel-n64-gcc15.patch
+++ b/pkgs/applications/emulators/libretro/cores/patches/parallel-n64-gcc15.patch
@@ -1,0 +1,62 @@
+From 3a891c0450c4968afbe76c594c22347354bd8834 Mon Sep 17 00:00:00 2001
+From: Moraxyc <i@qaq.li>
+Date: Sat, 3 Jan 2026 13:41:55 +0800
+Subject: [PATCH] Rename fsqrt to _fsqrt to avoid conflict
+
+---
+ mupen64plus-core/src/r4300/hacktarux_dynarec/assemble.h   | 2 +-
+ .../src/r4300/hacktarux_dynarec/hacktarux_dynarec.c       | 8 ++++----
+ 2 files changed, 5 insertions(+), 5 deletions(-)
+
+diff --git a/mupen64plus-core/src/r4300/hacktarux_dynarec/assemble.h b/mupen64plus-core/src/r4300/hacktarux_dynarec/assemble.h
+index 8161292..e256dd9 100644
+--- a/mupen64plus-core/src/r4300/hacktarux_dynarec/assemble.h
++++ b/mupen64plus-core/src/r4300/hacktarux_dynarec/assemble.h
+@@ -1650,7 +1650,7 @@ static INLINE void fchs(void)
+ }
+ 
+ 
+-static INLINE void fsqrt(void)
++static INLINE void _fsqrt(void)
+ {
+    put8(0xD9);
+    put8(0xFA);
+diff --git a/mupen64plus-core/src/r4300/hacktarux_dynarec/hacktarux_dynarec.c b/mupen64plus-core/src/r4300/hacktarux_dynarec/hacktarux_dynarec.c
+index 89fb982..2fc70d1 100644
+--- a/mupen64plus-core/src/r4300/hacktarux_dynarec/hacktarux_dynarec.c
++++ b/mupen64plus-core/src/r4300/hacktarux_dynarec/hacktarux_dynarec.c
+@@ -6317,13 +6317,13 @@ void gensqrt_d(void)
+ #ifdef __x86_64__
+    mov_xreg64_m64rel(RAX, (uint64_t *)(&reg_cop1_double[dst->f.cf.fs]));
+    fld_preg64_qword(RAX);
+-   fsqrt();
++   _fsqrt();
+    mov_xreg64_m64rel(RAX, (uint64_t *)(&reg_cop1_double[dst->f.cf.fd]));
+    fstp_preg64_qword(RAX);
+ #else
+    mov_eax_memoffs32((unsigned int *)(&reg_cop1_double[dst->f.cf.fs]));
+    fld_preg32_qword(EAX);
+-   fsqrt();
++   _fsqrt();
+    mov_eax_memoffs32((unsigned int *)(&reg_cop1_double[dst->f.cf.fd]));
+    fstp_preg32_qword(EAX);
+ #endif
+@@ -7301,13 +7301,13 @@ void gensqrt_s(void)
+ #ifdef __x86_64__
+    mov_xreg64_m64rel(RAX, (uint64_t *)(&reg_cop1_simple[dst->f.cf.fs]));
+    fld_preg64_dword(RAX);
+-   fsqrt();
++   _fsqrt();
+    mov_xreg64_m64rel(RAX, (uint64_t *)(&reg_cop1_simple[dst->f.cf.fd]));
+    fstp_preg64_dword(RAX);
+ #else
+    mov_eax_memoffs32((unsigned int *)(&reg_cop1_simple[dst->f.cf.fs]));
+    fld_preg32_dword(EAX);
+-   fsqrt();
++   _fsqrt();
+    mov_eax_memoffs32((unsigned int *)(&reg_cop1_simple[dst->f.cf.fd]));
+    fstp_preg32_dword(EAX);
+ #endif
+-- 
+2.52.0
+

